### PR TITLE
feat: add fmeval-specific user agent header to botocore config for telemetry purposes 

### PR DESCRIPTION
**Pull request by danielezhu**
*Description of changes:*
This PR implements very basic telemetry through the use of user agent headers that get included when making SageMaker `DescribeEndpoint` API calls.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -90,6 +100,29 @@ Once you have created a virtual environment with python3.10, run the following c
...
+
+  File "<string>", line 923, in main
+  File "<string>", line 562, in run
+```
+then you will need to tweak the poetry installation script and re-run it.
+
+Steps:
+1. `curl -sSL https://install.python-poetry.org > poetry_script.py`
+2. Change the `symlinks` argument in `builder = venv.EnvBuilder(clear=True, with_pip=True, symlinks=False)` to `True`. See mionker's comment [here](https://github.com/python-poetry/install.python-poetry.org/issues/56) for an explanation.
+3. `python poetry_script.py --version 1.2.0` (where `1.2.0` is the version listed in `devtool`; this may change after the time of this writing).
```
Comment (keerthanvasist) [inline README.md:119]: Currently, we have already updated this to 1.8.2

**Review by keerthanvasist — APPROVED** [review]

**Review by keerthanvasist — APPROVED** [review]

**Review by oyangz — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -60,6 +60,16 @@ eval_output = eval_algo.evaluate(model=model_runner, dataset_config=config)
 [examples](https://github.com/aws/fmeval/tree/main/examples) for more details around the usage of
 eval algorithms.*
 
+## Telemetry
+By default, `fmeval` has very basic telemetry enabled, purely for tracking usage of the package.
+The only data that is tracked is the number of SageMaker, JumpStart, or Bedrock `ModelRunner` objects that get created.
```
Comment (malhotra18) [inline README.md:65]: Correction, we are not doing it for bedrock.

File: README.md [hunk 3]
```diff
@@ -60,6 +60,16 @@ eval_output = eval_algo.evaluate(model=model_runner, dataset_config=config)
 [examples](https://github.com/aws/fmeval/tree/main/examples) for more details around the usage of
 eval algorithms.*
 
+## Telemetry
+By default, `fmeval` has very basic telemetry enabled, purely for tracking usage of the package.
```
Comment (malhotra18) [inline README.md:64]: Let's rephrase this to below:
```
`fmeval` has a telemetry enabled, for tacking the usage of AWS provided/hosted LLMs.
This data is tracked using the number of SageMaker, JumpStart, or Bedrock `ModelRunner` objects that get created. 
This can be disabled by setting the `DISABLE_FMEVAL_TELEMETRY` as `false`.
```

Question:
* Are we doing this for Bedrock? If not, then let's remove it.

File: src/fmeval/model_runners/util.py [hunk 4]
```diff
@@ -9,12 +9,32 @@
...
 
+
 logger = logging.getLogger(__name__)
 
 
+def get_user_agent_extra() -> str:
+    """
+    :return: A string to be used as the user_agent_extra parameter in a botocore config.
+    """
+    # Obtain user-agent headers for information such as SageMaker notebook instance type and SageMaker Studio app type.
```
Comment (malhotra18) [inline src/fmeval/model_runners/util.py:25]: where is this information coming from? `determine_prefix`? If yes, then can we make it more apparent in docstrings. 

And `determine_prefix` will give us all the headers. right? Not just the ones added by sagemaker_python_sdk.

File: README.md (hunk 3 above)
Comment (danielezhu) [reply README.md:64]: We're not doing this for Bedrock; that was my mistake. Will correct this alongside the rest of the README edits.

File: src/fmeval/model_runners/util.py (hunk 4 above)
Comment (danielezhu) [reply src/fmeval/model_runners/util.py:25]: `determine_prefix` is a [util function imported from the SageMaker Python SDK](https://github.com/aws/sagemaker-python-sdk/blob/2a52478e7fd185f197e944568173c58bd2495d78/src/sagemaker/user_agent.py#L66C5-L93). It returns only python SDK-related headers, and not all headers.

Our code is using the `user_agent_extra` parameter to pass the SDK headers and our own `fmeval` header as "extras", which get added to whatever headers are already present in the user agent.
